### DISCUSSION TITLE
running the AmpliPi in docker-pi?

### DISCUSSION THREAD

**Discussion by marc-portier** [micro-nova/AmpliPi#208] [Q&A]
Was wondering if any one out there has been trying to run the AmpliPi stack in a docker setup.
Or any other straightforward to have it around virtually, locally and isolated?

Main goal is to be able to test any associated integration development.

Is there interest to having such thing?

**Comment by linknum23** [comment]
Have you checked out the https://github.com/micro-nova/AmpliPi/blob/main/docs/DEVELOPING.md guide? It could probably use a little polishing and organization, but most of the info you need is there to start developing/testing AmpliPi's software without a real AmpliPi. 

We have been thinking about making a docker compose for a developer setup, but haven't gotten around to it, mostly because it would take a little bit to figure out how to output audio from it, even if it is just one stereo output.

**Reply by marc-portier** [reply]
Thx.
No did not see that page earlier, so will check it out.  Even if I take the dockerpi route eventually that still seems to have solid info.  (I'll be sure to report back if I still consider it to be a useful approach - and of course others can chime in as well)

**Reply by marc-portier** [reply]
started giving this a spin: https://github.com/lukechilds/dockerpi

first attempt to pass through the sound doesn't fly well
asked for some extra hints --> https://github.com/lukechilds/dockerpi/issues/49

parked this approach for now, the local ```scripts/run_debug_server ``` on my ubuntu works nicely - can't test out the spotify stuff though

**Reply by linknum23** [reply]
For spottily support  I'm fairly certain you can just replace streams/vollibrespot with the vollibrespot in this archive https://github.com/ashthespy/Vollibrespot/releases/download/v0.2.4/vollibrespot-x86_64.tar.xz .

**Reply by linknum23** [reply]
Good luck using dockerpi! That could make it easy to generate verifiable pi images automatically for every release and make some of our testing a little more sophisticated.
